apps/rewards/app/components/Content/MyRewards.js

```diff
@@ -21,6 +24,39 @@ import { useAppState } from '@aragon/api-react'
 import BigNumber from 'bignumber.js'
 import { displayCurrency } from '../../utils/helpers'
 
+const PendingStatus = (theme) => (
+  <div style={{ display: 'flex' }}>
+    <IconClock style={{
+      marginRight: '4px',
+      marginTop: '-2px',
+      color: theme.warning,
+    }}/>
+    Pending
+  </div>
+)
+
+const ReadyStatus = (theme) => (
+  <div style={{ display: 'flex' }}>
+    <IconCheck style={{
+      marginRight: '4px',
+      marginTop: '-2px',
+      color: theme.positive,
+    }}/>
+    Ready to claim
+  </div>
+)
+
+const ClaimedStatus = (theme) => (
+  <div style={{ display: 'flex' }}>
+    <IconCircleCheck style={{
+      marginRight: '4px',
+      marginTop: '-2px',
```

Comment:
Do you think it's worth pulling these common styles into a separate component?

```jsx
const Status = ({ color, Icon, text }) => (
  <div css="display: flex; align-items: center">
    <Icon css={`margin-right: ${0.5 * GU}px; color: ${color}`} />
    {text}
  </div>
)
```

Now that I wrote this out, I want to note two things and ask another.

1. Having a component that receives _an argument_ (`theme`) rather than _props_ (`{ theme }`) seems unexpected and error-prone to me
2. We can re-define `theme` within this component. `const theme = useTheme()`. Even though it means we need to switch from an implicit to an explicit return, I think the simplification of the calling code makes it worth it.
2. Does the `align-items: center` mean we can get rid of the `margin-top`?

Moving on, the above can be used as:

```jsx
<Status color={theme.positive} Icon={IconCircleCheck} text="Claimed" />
```

Now that I've written this out, I'm not sure if it's even worth having intermediate components for each kind of status, although it does keep the consuming code concise. What do you think?

Reply:
Thinking through this and how we'll pass `color` in but keep `theme` within the `Status` component, maybe something like this makes sense:

```jsx
const types = {
  claimed: { icon: IconCircleCheck, text: 'Claimed', color: 'positive' },
  ready: { icon: IconCheck, text: 'Ready to claim', color: 'positive' },
  pending: { icon: IconClock, text: 'Pending', color: 'warning' },
}

const Status = ({ type }) => {
  const theme = useTheme()
  const status = types[type]
  const text = status.text
  const Icon = status.icon
  const color = theme[status.color]
  return (
    <div css="display: flex; align-items: center">
      <Icon css={`margin-right: ${0.5 * GU}px; color: ${color}`} />
      {text}
    </div>
  )
}

Status.propTypes = {
  type: PropTypes.oneOf(Object.keys(types)).isRequired,
}

Reply:
@chadoh, thank you so much for your review. As always, very instructive. Regarding your specific points:

1. Thank you for explaining this to me. It wasn't clear why that component wasn't really working as a component, but rather as a function.
2. Is it the same in terms of performance? Aren't we creating multiple `theme` instances? Are hooks supposed to be used like that?
3. Nice try, but no. The icon needs to be a little up in order to match the text:
![Screenshot from 2019-11-15 14-29-47](https://user-images.githubusercontent.com/16065447/68971234-13c04a00-07b7-11ea-8350-1f9b06f3819c.png)
Otherwise, if it's in the center, the text will be higher:
![Screenshot from 2019-11-15 14-30-23](https://user-images.githubusercontent.com/16065447/68971260-22a6fc80-07b7-11ea-9612-19aa89deab76.png)
But my takeaway from this is that you make this sort of reviews by just writing the text on GitHub without any kind of testing, which I find quite impressive.

Reply:
I think your solution for having a single `Status` component with a single `type` prop is brilliant.